kaizen/cli/commands/test.py

```
@@ -411,21 +413,235 @@ def _save_summary_report(console: Console, test_result: TestResult, config: Any)
+            for i, attempt_data in enumerate(test_result.test_attempts):
+                console.print(f"[dim]DEBUG: Processing attempt {i+1}, type: {type(attempt_data)}[/dim]")
+                try:
+                    if isinstance(attempt_data, dict):
+                        # Handle memory system attempts (which have test_execution_result: None)
+                        test_cases = []
+                        test_results = attempt_data.get('test_results', {})
+                        # Debug print the structure of test_results
+                        console.print(f"[dim]DEBUG: test_results structure for attempt {attempt_data.get('attempt_number', 'unknown')}: {repr(test_results)[:300]}[/dim]")
+                        # Try to extract test cases from legacy format
+                        if 'tests' in test_results and isinstance(test_results['tests'], dict):
+                            console.print(f"[dim]DEBUG: Found 'tests' key in test_results[/dim]")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)